packages/@aws-cdk/aws-stepfunctions/lib/state-machine.ts

```diff
@@ -148,6 +148,87 @@ abstract class StateMachineBase extends Resource implements IStateMachine {
       resourceArns: [this.stateMachineArn],
     });
   }
+
+  /**
+   * Grant the given identity permissions to read results from state
+   * machine.
+   */
+  public grantRead(identity: iam.IGrantable): iam.Grant {
+    iam.Grant.addToPrincipal({
+      grantee: identity,
+      actions: [
+        'states:ListExecutions',
+        'states:ListStateMachines',
+      ],
+      resourceArns: [this.stateMachineArn],
+    });
+    iam.Grant.addToPrincipal({
+      grantee: identity,
+      actions: [
+        'states:DescribeExecution',
+        'states:DescribeStateMachineForExecution',
+        'states:GetExecutionHistory',
+      ],
+      resourceArns: [`${this.executionArn()}:*`],
+    });
+    return iam.Grant.addToPrincipal({
+      grantee: identity,
+      actions: [
+        'states:ListActivities',
+        'states:DescribeStateMachine',
+        'states:DescribeActivity',
+      ],
+      resourceArns: ['*'],
+    });
+  }
+
+  /**
+   * Grant the given identity task response permissions for a resource
+   *
+   * @default stateMachineArn
+   */
+  public grantTaskResponse(identity: iam.IGrantable, activityArn?: string): iam.Grant {
+    // Validate activityArn
+    if (activityArn && Arn.parse(activityArn, ':').resource !== 'activity') {
+      throw new Error('activityArn must be a valid activity ARN');
```

Comment:
I like to say that Error messages should he "help" messages. Rephrase to say:
* error - invalid activity arn
* how to reconcile - expected format
* what we received - parameter supplied

Reply:
no more error messages